File: frontend/src/services/models/Category.ts
export enum CategoryType {
    CITY = "CITY",
    INTERESTS = "INTERESTS",
    TARGET = "TARGET",
    SKILL = "SKILL",
    CATEGORY = "CATEGORY"
}

export const categoryTypeList: Record<string, string> = {
    [CategoryType.CITY]: "Город",
    [CategoryType.INTERESTS]: "Интерес",
    [CategoryType.TARGET]: "Цель",
    [CategoryType.SKILL]: "Навык",
    [CategoryType.CATEGORY]: "Категория",
}

export interface Category {
    id: number;
    name: string;
    color: string;
    icon: string;

    isCustomize: boolean;

    type: CategoryType;
}

File: backend/category-ms/src/main/java/ru/denis/category/models/Category.java
package ru.denis.category.models;

import jakarta.persistence.*;
import lombok.Data;
import ru.denis.category.CategoryType;

@Entity
@Data
@Table(name = "categories")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 25)
    private String name;

    @Column(length = 6)
    private String color = "FFFFFF";
    @Lob
    @Column(columnDefinition = "TEXT")
    private String icon = "";
    private Boolean isCustomize = true;
    private Boolean isModerated = false;

    @Enumerated(EnumType.STRING)
    private CategoryType type;
}
